libtelemetry/src/test/java/com/mapbox/android/telemetry/MapboxTelemetryTest.java

```diff
@@ -473,6 +473,38 @@ public void checksSessionIdRotationIntervalNotUpdated() throws Exception {
     assertFalse(notUpdatedSessionInterval);
   }
 
+  @Test
+  public void checksAppUserTurnstileNotQueued() throws Exception {
+    Context mockedContext = obtainNetworkConnectedMockedContext();
+    TelemetryClient mockedTelemetryClient = mock(TelemetryClient.class);
+    Callback mockedHttpCallback = mock(Callback.class);
+    EventsQueue eventsQueue = new EventsQueue(mock(FlushQueueCallback.class));
+    MapboxTelemetry theMapboxTelemetry = obtainMapboxTelemetryWith(mockedContext, mockedTelemetryClient,
+      mockedHttpCallback, eventsQueue);
+
+    Event whitelistedEvent = new AppUserTurnstile("anySdkIdentifier", "anySdkVersion", false);
+    theMapboxTelemetry.enable();
+    theMapboxTelemetry.push(whitelistedEvent);
+
+    assertEquals(0, theMapboxTelemetry.getQueue().queue.size());
```

Comment:
you could also do `assertTrue(theMapboxTelemetry.isQueueEmpty())` if you'd expose `boolean isQueueEmpty()` api.